sqlmesh/core/config/connection.py

```diff
@@ -1350,6 +1350,50 @@ def _static_connection_kwargs(self) -> t.Dict[str, t.Any]:
         }
 
 
+class ClickhouseConnectionConfig(ConnectionConfig):
+    """
+    Clickhouse Connection Configuration.
+
+    Property reference: https://clickhouse.com/docs/en/integrations/python#client-initialization
+    """
+
+    host: str
+    username: str
+    password: t.Optional[str] = None
+    port: t.Optional[int] = None
+    default_cluster: str = "default"
```

Comment:
Just confirming: This is the default name Clickhouse assigns if I don't provide a name? Basically just making sure `default` is a reasonable default to use or if we should require this field.

Reply:
My understanding is that if you stand up a zookeeper instance for managing the cluster, it will automatically create a cluster named `default`.

@erindru Is that right?

Reply:
Not quite. Basically there's always a cluster called "default" if you do `select * from system.clusters` but typically you'd give your cluster an actual name when you configure one. This results in two entries in `system.clusters`, one for "default" and one for your actual cluster.

Defaulting this to "default" would make queries work but it also might be surprising for users who were expecting it to pick their configured cluster. Perhaps leaving it unset by default and then throwing an exception if it's needed at runtime (ie, when running in cluster mode) to prompt the user to set it would be better?

Or if it's unset, querying `system.clusters` and if there's only one use that, otherwise raise an exception asking the user to set a default one?

Reply:
> Perhaps leaving it unset by default and then throwing an exception if it's needed at runtime (ie, when running in cluster mode) to prompt the user to set it would be better?

That sounds good

Reply:
Oh yeah - it also occurred to me that someone might have a cluster available but want to run in standalone mode. Currently. we will automatically use the cluster if it's detected.

What do you think about adding a `force_standalone_mode` arg to the connection so people can bypass the cluster if needed?

Reply:
I guess since I've not used Clickhouse before I'm not familiar with the different ways people use it. 

Another way to do this could be to assume standalone mode if the `cluster` property isn't set, rather than throwing an error or having another property to force standalone mode.

Rather than trying to guess all the different usage combinations up front we could also just let them naturally come out of the woodwork when people start using this feature